plugins/modules/ec2_asg.py

```diff
@@ -1523,6 +1544,40 @@ def replace(connection):
     return changed, asg_properties
 
 
+def detach(connection):
+    group_name = module.params.get('name')
+    detach_instances = module.params.get('detach_instances')
+    as_group = describe_autoscaling_groups(connection, group_name)[0]
```

Comment:
is there a case where multiple autoscaling groups are returned here?

Reply:
I think as we are passing only single group name, we will only single auto scaling group's info.

Reply:
just checking! Wasnt sure if there were cases where multiple groups had the same name

Reply:
ASG name is unique within a region and account.  Since we can only act on one region and account at a time, this works well.